### DISCUSSION TITLE
link problem in the authentication-tutorial.md for the /app/login need to change the path information.

### DISCUSSION THREAD

**Discussion by knupel** [gatsbyjs/gatsby#28347] [Help]
## Description

I'm not sure is a bug or miss understanding from my side. I make the tutorial for the authentication. But at the end when do `gatsby develop` I receive from my browser is path problem when I click on `login` or `profile` the app try to go `/app/login`or `/app/profile` but in code snippet all the time the path indicate is `/app/profile` or `/app/login`so that's make sens  but it's ask to to create the file in the path `src/components/profile.js`and `src/components/login.js`

 plus in the code of `app.js` the import file is 
```
import Profile from "../components/profile"
import Login from "../components/login"
```

To solve it  : 
I put the the file `profile.js`and `login.js` and create a folder `app` and change the import path 

```
import Profile from "../app/profile"
import Login from "../app/login"
```

Now that's work great.

**Comment by LekoArts** [comment]
The `src/pages/app.js` file explained in [this section](https://www.gatsbyjs.com/tutorial/authentication-tutorial/#creating-client-only-routes) is handling the routing. Changing the import paths isn't the source of your problem.

If you can link your repository with the code someone could look over it to spot why your problem occurs. While it's unlikely that the tutorial is incorrect, we want to double check.
